src/compiler/scala/tools/nsc/typechecker/Unapplies.scala

```diff
@@ -60,7 +60,8 @@ trait Unapplies extends ast.TreeDSL {
   }
 
   private def constrTparamsInvariant(cdef: ClassDef): List[TypeDef] = {
-    val ClassDef(_, _, tparams, _) = resetAttrs(cdef.duplicate)
+    val prunedClassDef = deriveClassDef(cdef)(tmpl => deriveTemplate(tmpl)(_ => Nil))
```

Comment:
Why is this different than above, where you wrote `deriveClassDef(cdef)(_ => Template(Nil, noSelfType, Nil))`?